prompt/prompt-executor/prompt-executor-clients/prompt-executor-bedrock-client/src/jvmMain/kotlin/ai/koog/prompt/executor/clients/bedrock/BedrockLLMClient.kt

```diff
@@ -0,0 +1,292 @@
+
+    /**
+     * Creates a new Bedrock LLM client configured with the specified AWS credentials and settings.
+     *
+     * @param awsAccessKeyId The AWS access key ID for authentication
+     * @param awsSecretAccessKey The AWS secret access key for authentication
+     * @param settings Configuration settings for the Bedrock client, such as region and endpoint
+     * @param clock A clock used for time-based operations
+     * @return A configured [LLMClient] instance for Bedrock
+     */
+    public constructor(
+        awsAccessKeyId: String,
+        awsSecretAccessKey: String,
+        settings: BedrockClientSettings = BedrockClientSettings(),
+        clock: Clock = Clock.System
+    ) : this(
+        bedrockClient = BedrockRuntimeClient {
+            this.region = settings.region
+            this.credentialsProvider = StaticCredentialsProvider {
+                this.accessKeyId = awsAccessKeyId
+                this.secretAccessKey = awsSecretAccessKey
+            }
+
+            // Configure custom endpoint if provided
+            settings.endpointUrl?.let { url ->
+                this.endpointUrl = Url.parse(url)
+            }
+
+            // Configure retry policy
+            this.retryStrategy = StandardRetryStrategy {
+                maxAttempts = settings.maxRetries
+            }
+        },
+        clock = clock
+    )
+
+    private val json = Json {
+        ignoreUnknownKeys = true
+        isLenient = true
+        explicitNulls = false
+    }
+
+    internal fun getBedrockModelFamily(model: LLModel): BedrockModelFamilies {
+        require(model.provider == LLMProvider.Bedrock) { "Model ${model.id} is not a Bedrock model" }
+        return when {
+            model.id.startsWith("anthropic.claude") -> BedrockModelFamilies.AnthropicClaude
+            model.id.startsWith("amazon.nova") -> BedrockModelFamilies.AmazonNova
+            model.id.startsWith("ai21.jamba") -> BedrockModelFamilies.AI21Jamba
+            model.id.startsWith("meta.llama") -> BedrockModelFamilies.Meta
+            else -> throw IllegalArgumentException("Model ${model.id} is not a supported Bedrock model")
+        }
+    }
+
+    override suspend fun execute(
+        prompt: Prompt,
+        model: LLModel,
+        tools: List<ToolDescriptor>
+    ): List<Message.Response> {
+        logger.debug { "Executing prompt for model: ${model.id}" }
+
+        val modelFamily = getBedrockModelFamily(model)
+        require(model.capabilities.contains(LLMCapability.Completion)) {
+            "Model ${model.id} does not support chat completions"
+        }
+
+        // Check tool support
+        if (tools.isNotEmpty() && !model.capabilities.contains(LLMCapability.Tools)) {
+            throw IllegalArgumentException("Model ${model.id} does not support tools")
+        }
+
+        val requestBody = when (modelFamily) {
+            is BedrockModelFamilies.AI21Jamba ->
+                if (model.id.startsWith("ai21.jamba")) {
```

Comment:
Looks like these if's here and below are redundant, since this check is already performed in `getBedrockModelFamily`